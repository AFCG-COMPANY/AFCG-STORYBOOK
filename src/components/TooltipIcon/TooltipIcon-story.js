import React from 'react';
import { storiesOf } from '@storybook/react';
import TooltipIcon from '../TooltipIcon';

storiesOf('TooltipIcon', module)
  .addWithInfo(
    'default',
    `
    Tooltip Icon
    `,
    () => (
      <TooltipIcon tooltipText="Filter">
        <svg width="16" height="12" viewBox="0 0 16 12">
          <g fillRule="nonzero">
            <path d="M8.05 2a2.5 2.5 0 0 1 4.9 0H16v1h-3.05a2.5 2.5 0 0 1-4.9 0H0V2h8.05zm2.45 2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3zM3.05 9a2.5 2.5 0 0 1 4.9 0H16v1H7.95a2.5 2.5 0 0 1-4.9 0H0V9h3.05zm2.45 2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3z" />
          </g>
        </svg>
      </TooltipIcon>
    )
  )
  .addWithInfo(
    'with custom direction',
    `
    Definition Tooltip with custom direction
    `,
    () => (
      <div style={{ marginTop: '2rem' }}>
        <TooltipIcon direction="top" tooltipText="Filter">
          <svg width="16" height="12" viewBox="0 0 16 12">
            <g fillRule="nonzero">
              <path d="M8.05 2a2.5 2.5 0 0 1 4.9 0H16v1h-3.05a2.5 2.5 0 0 1-4.9 0H0V2h8.05zm2.45 2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3zM3.05 9a2.5 2.5 0 0 1 4.9 0H16v1H7.95a2.5 2.5 0 0 1-4.9 0H0V9h3.05zm2.45 2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3z" />
            </g>
          </svg>
        </TooltipIcon>
      </div>
    )
  );
