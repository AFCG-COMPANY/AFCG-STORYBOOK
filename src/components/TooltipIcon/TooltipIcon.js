import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

const TooltipIcon = ({
  className,
  children,
  direction,
  tooltipText,
  ...rest
}) => {
  const tooltipClassName = cx({
    [className]: !!className,
    'bx--tooltip-icon': true,
  });
  const triggerClassName = cx({
    'bx--tooltip__trigger': true,
    'bx--tooltip--icon__bottom': direction === 'bottom',
    'bx--tooltip--icon__top': direction === 'top',
  });
  return (
    <div {...rest} className={tooltipClassName}>
      <button className={triggerClassName} title={tooltipText}>
        {children}
      </button>
    </div>
  );
};

TooltipIcon.propTypes = {
  /**
   * Specify an icon as children that will be used as the tooltip trigger. This
   * can be an icon from our Icon component, or a custom SVG element.
   */
  children: PropTypes.node.isRequired,

  /**
   * Specify the direction of the tooltip. Can be either bottom or top.
   */
  direction: PropTypes.oneOf(['bottom', 'top']).isRequired,

  /**
   * Provide the text that will be displayed in the tooltip when it is rendered.
   */
  tooltipText: PropTypes.node.isRequired,
};

TooltipIcon.defaultProps = {
  direction: 'bottom',
};

export default TooltipIcon;
