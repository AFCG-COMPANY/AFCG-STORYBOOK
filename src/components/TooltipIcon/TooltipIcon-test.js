import React from 'react';
import { mount } from 'enzyme';
import TooltipIcon from '../TooltipIcon';

describe('TooltipIcon', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      direction: 'bottom',
      children: <svg />,
      className: 'custom-class',
      tooltipText: 'tooltip text',
    };
  });

  it('should render', () => {
    const wrapper = mount(<TooltipIcon {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow the user to specify the direction', () => {
    const wrapper = mount(<TooltipIcon {...mockProps} direction="top" />);
    expect(wrapper).toMatchSnapshot();
  });
});
