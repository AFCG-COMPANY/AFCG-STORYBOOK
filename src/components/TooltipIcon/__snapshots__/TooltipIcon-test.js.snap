// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipIcon should allow the user to specify the direction 1`] = `
<TooltipIcon
  className="custom-class"
  direction="top"
  tooltipText="tooltip text"
>
  <div
    className="custom-class bx--tooltip-icon"
  >
    <button
      className="bx--tooltip__trigger bx--tooltip--icon__top"
      title="tooltip text"
    >
      <svg />
    </button>
  </div>
</TooltipIcon>
`;

exports[`TooltipIcon should render 1`] = `
<TooltipIcon
  className="custom-class"
  direction="bottom"
  tooltipText="tooltip text"
>
  <div
    className="custom-class bx--tooltip-icon"
  >
    <button
      className="bx--tooltip__trigger bx--tooltip--icon__bottom"
      title="tooltip text"
    >
      <svg />
    </button>
  </div>
</TooltipIcon>
`;
